src/test/regress/expected/with_transactions.out

```diff
@@ -0,0 +1,102 @@
+SELECT create_distributed_table('second_raw_table', 'tenant_id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+INSERT INTO 
+	raw_table (tenant_id, income, created_at) 
+SELECT 
+	i % 10, i * 10.0, timestamp '2014-01-10 20:00:00' + i * interval '1 day' 
+FROM 
+	generate_series (0, 100) i;
+INSERT INTO second_raw_table SELECT * FROM raw_table;
+SET client_min_messages TO DEBUG1;
+-- run a transaction which DELETE 
+BEGIN;
+	WITH ids_to_delete AS
+		(
+			SELECT tenant_id FROM raw_table WHERE income < 250
+		),
+		deleted_ids AS
+		(
+			DELETE FROM raw_table WHERE created_at  < '2014-02-10 20:00:00' AND tenant_id IN (SELECT * from ids_to_delete) RETURNING tenant_id
+		)
+		UPDATE raw_table SET income = income * 2 WHERE tenant_id IN (SELECT tenant_id FROM deleted_ids);
+DEBUG:  common table expressions are not supported in distributed modifications
+DEBUG:  generating subplan 3_1 for CTE ids_to_delete: SELECT tenant_id FROM with_transactions.raw_table WHERE (income < (250)::double precision)
+DEBUG:  generating subplan 3_2 for CTE deleted_ids: DELETE FROM with_transactions.raw_table WHERE ((created_at < 'Mon Feb 10 20:00:00 2014 PST'::timestamp with time zone) AND (tenant_id IN (SELECT ids_to_delete.tenant_id FROM (SELECT intermediate_result.tenant_id FROM read_intermediate_result('3_1'::text, 'binary'::citus_copy_format) intermediate_result(tenant_id integer)) ids_to_delete))) RETURNING tenant_id
+DEBUG:  Plan 3 query after replacing subqueries and CTEs: UPDATE with_transactions.raw_table SET income = (income * (2)::double precision) WHERE (tenant_id IN (SELECT deleted_ids.tenant_id FROM (SELECT intermediate_result.tenant_id FROM read_intermediate_result('3_2'::text, 'binary'::citus_copy_format) intermediate_result(tenant_id integer)) deleted_ids))
+ROLLBACK;
+-- see that both UPDATE and DELETE commands are rollbacked
+SELECT count(*) FROM raw_table;
+ count 
+-------
+   101
+(1 row)
+
+SELECT max(income) FROM raw_table;
+ max  
+------
+ 1000
+(1 row)
+
+-- multi-statement multi shard modifying statements should work
+BEGIN;
+	WITH ids_inserted AS
+		(
+			INSERT INTO raw_table VALUES (11, 1000, now()) RETURNING tenant_id 
+		)
+		UPDATE raw_table SET created_at = '2001-02-10 20:00:00' WHERE tenant_id IN (SELECT tenant_id FROM ids_inserted);
+DEBUG:  common table expressions are not supported in distributed modifications
+DEBUG:  generating subplan 8_1 for CTE ids_inserted: INSERT INTO with_transactions.raw_table (tenant_id, income, created_at) VALUES (11, 1000, now()) RETURNING tenant_id
+DEBUG:  Plan 8 query after replacing subqueries and CTEs: UPDATE with_transactions.raw_table SET created_at = 'Sat Feb 10 20:00:00 2001 PST'::timestamp with time zone WHERE (tenant_id IN (SELECT ids_inserted.tenant_id FROM (SELECT intermediate_result.tenant_id FROM read_intermediate_result('8_1'::text, 'binary'::citus_copy_format) intermediate_result(tenant_id integer)) ids_inserted))
+	TRUNCATE second_raw_table;
```

Comment:
I realized that we've missed running `SELECT`s inside the transactions.

Could we add the following two  (or any other queries) `SELECT` statements in this transction:
```SQL

SELECT count (*) FROM second_raw_table;
SELECT count(DISTINCT created_at) FROM raw_table;
```